### PR TITLE
added all math equations for loot variables /drop

### PR DESCRIPTION
**Pull request by Noanan**
fix the random seeding

### CODE REVIEW

File: pkg/infra/game/math/math.go [hunk 1]
```diff
@@ -8,18 +8,21 @@ import (
 	"infra/config"
 )
 
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
```
Comment (Razvan00Rusu) [inline pkg/infra/game/math/math.go:14]: I might just not be seeing it on this PR but is init function being called somewhere?

Comment (antonypap) [reply pkg/infra/game/math/math.go:14]: Check [here](https://www.developer.com/languages/inti-function-golang/)

File: pkg/infra/main_helper.go [hunk 2]
```diff
@@ -175,12 +174,14 @@ func checkHpPool() bool {
 	return false
 }
 
-func generateLootPool(numAgents int, currentLevel uint) *state.LootPool {
-	makeItems := func() *commons.ImmutableList[state.Item] {
-		nItems := rand.Intn(numAgents) / 10
+func generateLootPool(numAgents int) *state.LootPool {
```
Comment (neeldug) [inline pkg/infra/main_helper.go:177]:
```suggestion
func generateLootPool(numAgents uint) *state.LootPool {
	nWeapons, nShields := gamemath.GetEquipmentDistribution(numAgents)
	nHealthPotions, nStaminaPotions := gamemath.GetPotionDistribution(numAgents)

	makeItems := func(nItems uint, stats uint) *commons.ImmutableList[state.Item] {
		items := make([]state.Item, nItems)
		for i := uint(0); i < nItems; i++ {
			items[i] = *state.NewItem(uuid.NewString(), stats)
		}
		sort.SliceStable(items, func(i, j int) bool {
			return items[i].Value() > items[j].Value()
		})
		return commons.NewImmutableList(items)
	}

	return state.NewLootPool(
		// Weapons
		makeItems(nWeapons, gamemath.GetWeaponDamage(globalState.MonsterHealth, numAgents)),
		// Shields
		makeItems(nShields, gamemath.GetShieldProtection(globalState.MonsterAttack, numAgents)),
		// Health Potions
		makeItems(nHealthPotions, gamemath.GetHealthPotionValue(globalState.MonsterAttack, numAgents)),
		// Stamina Potions
		makeItems(nStaminaPotions, gamemath.GetStaminaPotionValue(globalState.MonsterHealth, numAgents)),
	)
}
```

File: pkg/infra/main.go [hunk 3]
```diff
@@ -165,7 +165,7 @@ func startGameLoop() {
 
 		// TODO: Loot Discussion Stage
 
-		lootPool := generateLootPool(len(agentMap), globalState.CurrentLevel)
+		lootPool := generateLootPool(len(agentMap))
```
Comment (neeldug) [inline pkg/infra/main.go:168]:
```suggestion
		lootPool := generateLootPool(uint(len(agentMap)))
```

**Review by Razvan00Rusu — APPROVED** [review]

**Pull request merged**